bridge_adaptivity/module/models.py

```diff
@@ -116,49 +140,42 @@ def get_absolute_url(self):
 class Engine(models.Model):
     """Defines engine settings."""
 
-    DEFAULT_ENGINE_NAME = 'mock'
+    DEFAULT_ENGINE = 'engine_mock'
+    DRIVER = None
 
-    name = models.CharField(default=DEFAULT_ENGINE_NAME, max_length=255)
+    engine = models.CharField(choices=ENGINES, default=DEFAULT_ENGINE, max_length=100)
+    engine_name = models.CharField(max_length=255, blank=True, null=True, unique=True)
     host = models.URLField(blank=True, null=True)
     token = models.CharField(max_length=255, blank=True, null=True)
-
-    _engines_cache = {}
+    is_default = fields.BooleanField(default=False, help_text=_("If checked Engine will be used as the default!"))
 
     class Meta:
         unique_together = ('host', 'token')
 
     def __str__(self):
-        return "Engine: {}".format(self.name)
+        return "Engine: {}".format(self.engine_name)
+
+    def save(self, *args, **kwargs):
+        utils.save_model_parameter_true_once(self, 'is_default')
+        super(Engine, self).save(*args, **kwargs)
 
     @classmethod
     def get_default_engine(cls):
-        return Engine.objects.get_or_create(name=cls.DEFAULT_ENGINE_NAME)
-
-    def make_engine(self):
-        if self.name == 'mock':
-            # mock engine takes no params
-            settings = {}
-        else:
-            settings = {
-                'host': self.host,
-                'token': self.token
-            }
-        key = (self.name, tuple(settings.items()))
-        if key in Engine._engines_cache:
-            # if engine already in cache - return it
-            return Engine._engines_cache.get(key)
-        # otherwise create Engine, put it in cache and return it
-        engine_module = importlib.import_module('module.engines.engine_{}'.format(self.name.lower()))
-        engine_template = 'Engine{}'
-        driver = getattr(
-            engine_module, engine_template.format(self.name.upper()),  # try uppercase
-            getattr(
-                engine_module,
-                engine_template.format(self.name.capitalize()),  # if not found - try capitalized
-            )
+        return (
```

Comment:
I would avoid parentheses and reformat this code in a such way:
```
return Engine.objects.filter(
    is_default=True
).first() or Engine.objects.get_or_create(
    engine=cls.DEFAULT_ENGINE, engine_name='Mock', is_default=True
)[0]
```

but probably we should discuss it...

Reply:
I would leave as it is now if you don't mind